### PR TITLE
tests: retry leader of region

### PR DESCRIPTION
**Pull request by overvenus**
After doing some `ConfChange`s(eg. `AddNode`), `leader_of_region` may not find leaders if it gets stale `store_ids`.

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

File: tests/raftstore/cluster.rs [hunk 1]
```diff
@@ -313,51 +313,27 @@ impl<T: Simulator> Cluster<T> {
...
-                }
-                if leader.is_none() {
-                    leader = l;
-                    count += 1;
-                } else if l == leader {
-                    count += 1;
+                let leader = self.query_leader(*store_id, region_id);
+                if let Some(l) = leader {
+                    self.leaders.insert(region_id, l);
+                    break 'retry;
```
Comment (BusyJay) [inline tests/raftstore/cluster.rs]: What if the cluster is isolated?

**Comment by siddontang** [comment]
ping @overvenus

**Comment by overvenus** [comment]
I am currently busy with other stuff, let's close it until I have time to continue.

**Pull request closed without merging**